README
NAME
    App::proxyforurl - An online proxy PAC file parser

VERSION
    0.05

DESCRIPTION
    App::proxyforurl is a web application that can be used to test proxy PAC
    files.

    The server side is used to serve the web page, but does also provide
    functionlity for resolving hostnames and checking if an IP is within a
    given net.

  Demo
    Check out <http://home.thorsen.pm/proxyforurl> for a running example.

  Missing features
    The client side PAC parser does not fully support "dateRange()",
    "timeRange()" or "weekdayRange()". These functions simply return true,
    no matter what the intput is.

SYNOPSIS
      $ proxyforurl --listen http://*:8080;

SEE ALSO
    *   <http://findproxyforurl.com/>

    *   <https://github.com/pacparser/pacparser>

  DISCLAIMER
    The parsing is done using good old "eval()" on the client side, which
    means that the pasted PAC file could in theory contain code which could
    steal cookies, inject alien JavaScript or do other harmful things.

    There is a safety net installed to prevent this from happening, but
    since this is software, there might be bugs.

COPYRIGHT AND LICENSE
    Copyright (C) 2014, Jan Henning Thorsen

    This program is free software, you can redistribute it and/or modify it
    under the terms of the Artistic License version 2.0.

AUTHOR
    Jan Henning Thorsen - "jhthorsen@cpan.org"



Changes
Revision history for perl distribution App-proxyforurl

1.02 Not Released
 - Skipping api.t tests if not developing

1.01 2023-02-23T15:03:00
 - Fix fetching correct URLs with PROXYFORURL_X_REQUEST_BASE

1.00 2023-02-23T14:25:43
 - Fix demo link in documentation
 - Fix dnsDomainIs('microsoft.com', 'ft.com')
 - Add support for PROXYFORURL_BRAND_NAME
 - Add support for PROXYFORURL_BRAND_URL
 - Add support for PROXYFORURL_TEMPLATES
 - Add support for PROXYFORURL_X_REQUEST_BASE
 - Changed GET /gethostbyname to POST /v1/gethostbyname
 - Changed GET /within to POST /v1/is-in-net
 - Changed default style to https://picocss.com
 - Rewrote JavaScript

0.12 2021-12-14T19:16:57+0900
 - Fix correct quotes for pure-min.css

0.11 2021-10-20T13:27:26+0900
 - Skip test for CPAN smokers

0.10 2021-10-06T08:11:17+0900
 - Need to check $? instead of $! if gethostbyname fails #5

0.09 2021-08-28T16:47:03+0200
 - Fix demo link in documentation #4

0.08 2021-06-17T09:10:12+0900
 - Add support for X-Request-Base
 - Compatible with Mojolicious 9.11

0.07 2018-03-03T15:33:27+0100
 - Fix pure-min.css URL
 - Fix loading application in tests

0.06 2016-12-11T09:53:45+0100
 - Fix dnsDomainIs() should not use a regex matcher #2

0.05 2016-11-13T18:02:19+0100
 - Fix dnsDomainIs() for domains containing dot "." #1

0.04 2016-04-17T11:34:41+0200
 - Fix dnsDomainIs() test

0.03 2015-08-10T15:47:47+0200
 - Fix isPlainHostName() test

0.02 2015-08-09T18:30:45+0200
 - Fix link to demo
 - More padding between form fields
 - Bigger editor

0.01 2015-08-09T18:10:51+0200
 - Add script/proxyforurl


Makefile.PL
# Generated by git-ship. See 'git-ship --man' for help or https://github.com/jhthorsen/app-git-ship
use utf8;
use ExtUtils::MakeMaker;
my %WriteMakefileArgs = (
  NAME           => 'App::proxyforurl',
  AUTHOR         => 'Jan Henning Thorsen <jhthorsen@cpan.org>',
  LICENSE        => 'artistic_2',
  ABSTRACT_FROM  => 'lib/App/proxyforurl.pm',
  VERSION_FROM   => 'lib/App/proxyforurl.pm',
  EXE_FILES      => [qw(script/proxyforurl)],
  OBJECT         => '',
  BUILD_REQUIRES => {},
  TEST_REQUIRES  => {'Test::More'  => '0.88'},
  PREREQ_PM      => {'Mojolicious' => '9.11', 'NetAddr::IP' => '4.00'},
  META_MERGE     => {
    'dynamic_config' => 0,
    'meta-spec'      => {version => 2},
    'resources'      => {
      bugtracker => {web => 'https://github.com/jhthorsen/app-proxyforurl/issues'},
      homepage   => 'https://github.com/jhthorsen/app-proxyforurl',
      repository => {
        type => 'git',
        url  => 'https://github.com/jhthorsen/app-proxyforurl.git',
        web  => 'https://github.com/jhthorsen/app-proxyforurl',
      },
    },
    'x_contributors' => ['Jan Henning Thorsen <jhthorsen@cpan.org>'],
  },
  test => {TESTS => (-e 'META.yml' ? 't/*.t' : 't/*.t xt/*.t')},
);

unless (eval { ExtUtils::MakeMaker->VERSION('6.63_03') }) {
  my $test_requires = delete $WriteMakefileArgs{TEST_REQUIRES};
  @{$WriteMakefileArgs{PREREQ_PM}}{keys %$test_requires} = values %$test_requires;
}

WriteMakefile(%WriteMakefileArgs);


lib/App/proxyforurl.pm
package App::proxyforurl;
use strict;
use warnings;

our $VERSION = '1.01';

1;

=encoding utf8

=head1 NAME

App::proxyforurl - An online proxy PAC file parser

=head1 DESCRIPTION

L<App::proxyforurl> is a web application that can be used to test proxy PAC
files.

The L<server side|Mojolicious> is used to serve the web page, but does also
provide functionality for resolving hostnames and checking if an IP is within
a given net.

=head2 Demo

Check out L<https://thorsen.pm/proxyforurl> for a running example.

=head2 Missing features

The client side PAC parser does not fully support C<dateRange()>, C<timeRange()>
or C<weekdayRange()>. These functions simply return true, no matter what the
input is.

=head1 SYNOPSIS

  $ proxyforurl --listen http://*:8080;

With optional environment variables:

  $ PROXYFORURL_TEMPLATES=/path/to/templates \
    PROXYFORURL_BRAND_NAME=Thorsen \
    PROXYFORURL_BRAND_URL=https://thorsen.pm \
    PROXYFORURL_X_REQUEST_BASE="https://thorsen.pm/proxyforurl" \
    proxyforurl --listen http://*:8080;

=over 2

=item * PROXYFORURL_TEMPLATES

Can be set to a custom directory to override templates.

=item * PROXYFORURL_BRAND_NAME, PROXYFORURL_BRAND_URL

Used to change the menu item text and URL.

=item * PROXYFORURL_X_REQUEST_BASE

Can be set to a custom request base, in case the app is not mounted at "/".

=back

=head1 SEE ALSO

=over 4

=item * L<https://findproxyforurl.com/>

=item * L<https://github.com/pacparser/pacparser>

=back

=head1 DISCLAIMER

The parsing is done using good old C<eval()> on the client side, which means
that the pasted PAC file could in theory contain code which could steal
cookies, inject alien JavaScript or do other harmful things.

There is a safety net installed to prevent this from happening, but since
this is software, there might be bugs.

=head1 COPYRIGHT AND LICENSE

Copyright (C) Jan Henning Thorsen

This program is free software, you can redistribute it and/or modify it under
the terms of the Artistic License version 2.0.

=head1 AUTHOR

Jan Henning Thorsen - C<jhthorsen@cpan.org>

=cut
